advanced-menu/advanced-menu.py

```diff
@@ -124,6 +141,9 @@ async def on_thread_ready(self, thread, creator, category, initial_message):
             dummyMessage.embeds = []
             dummyMessage.stickers = []
 
+            if self.config["block_until_done"]:
+                self.bot.config["blocked"][str(thread._recipient.id)] = "Dropdown creation"
```

Comment:
Using the block system is very counter intuitive. Your system also does not account for normal usage of the block system and can cause blocks to be undone, or can cause users to remain blocked on bot crashes.

Reply:
There was no other perceivable way to stop people from sending messages in the thread. If the user is normally blocked, then no thread will be created and therefore there would only be a very specific window in which I partly accounted for but apparently never finished. If the bot crashes, the block is only stored in memory and therefore the user will be unblocked as expected.